### ISSUE TITLE
WHERE expression dont work with non en-US symbols.

### ISSUE THREAD

**Issue by fe-c** [dotnet/efcore#4781]
It can't use unicode symbols in queries:

```
var userName = "фыва"; // Name with unicode symbols
var userByName = db.Users.FirstOrDefault(x => x.Name == userName ); // returns null!
var userById = db.Users.FirstOrDefault(x => x.Id == 2974 ); // return User with Name='фыва' - this user exists.
var check = userById.Name.Equals(userName); // check == true!
```

Perhaps this is due to it is not added to the N' before string literal in a generated query:

```
db.Users.FromSql("SELECT * FROM Users WHERE Name='фыва'").Count(); // returns 0
db.Users.FromSql("SELECT * FROM Users WHERE Name=N'фыва'").Count(); // returns 1
```

Using latest EntityFramework.MicrosoftSqlServer –Pre with mssqllocaldb on VS2015.

**Comment by fe-c** [comment]
With name 'fox' all works fine.

**Comment by divega** [comment]
If this is just about the lack of N prefix on string literals, it should be fixed in current bits after #4667.

**Comment by fe-c** [comment]
Perhaps this is what it is. Waiting for the next release, thanks!
